OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -128,9 +140,17 @@ public override Activity Tick(Actor self)
 
 			// Fix a problem when the airplane is send to resupply near the airport
 			landingProcedures.Add(new Fly(self, Target.FromPos(w3), WDist.Zero, new WDist(turnRadius / 2)));
-			landingProcedures.Add(new Land(self, Target.FromActor(dest)));
-			landingProcedures.Add(new ResupplyAircraft(self));
-			landingProcedures.Add(NextActivity);
+
+			if (ShouldLandAtBuilding(self, dest))
+			{
+				plane.MakeReservation(dest);
+
+				landingProcedures.Add(new Land(self, Target.FromActor(dest)));
+				landingProcedures.Add(new ResupplyAircraft(self));
+			}
+
+			if (!planeInfo.AbortOnResupply)
+				landingProcedures.Add(NextActivity);
```

Comment:
Please re-write this in such a way that calling this activity from the Lua API won't throw away the queue.